opengrok-indexer/src/main/java/org/opengrok/indexer/history/HistoryGuru.java

```diff
@@ -480,7 +478,25 @@ public boolean hasAnnotation(File file) {
      * @return collection of added repositories
      */
     public Collection<RepositoryInfo> addRepositories(File[] files) {
-        return addRepositories(files, env.getNestingMaximum(), 0, false);
+        ExecutorService executor = env.getIndexerParallelizer().getRepositorySearchExecutor();
+        List<Future<Collection<RepositoryInfo>>> futures = new ArrayList<>();
+        for (File file: files) {
+            futures.add(executor.submit(() -> addRepositories(new File[]{file},
+                    env.getNestingMaximum(), 0, false)));
+        }
+
+        List<RepositoryInfo> repoList = new ArrayList<>();
+        futures.forEach(future -> {
+            try {
+                repoList.addAll(future.get());
+            } catch (Exception e) {
+                LOGGER.log(Level.WARNING, "failed to get future");
+            }
+        });
+
+        env.getIndexerParallelizer().bounceRepositorySearchExecutor();
```

Comment:
I wouldn't think a bounce would be required if futures are being awaited. Why is it necessary?

Reply:
just didn't want to keep the threads around.